### PR TITLE
[agent] add region support

### PR DESCRIPTION
**Pull request by gjc13**
The PR adds region support which will determine the preferred channels
and the allowed channels when forming or joining a Thread network.

- Add region and channel mask logic.
- Add region to command line arguments.
- Add d-bus get and set api.

### CODE REVIEW

File: src/common/region_code.hpp [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    kRegionCA      = 1, // < Canada
+    kRegionUS      = 2, // < United States
+    kRegionUnknown = 3, // < Unknown
+};
+
+RegionCode StringToRegionCode(const char *aRegionString);
+
+const char *RegionCodeToString(RegionCode aRegionCode);
+
+uint32_t GetSupportedChannelMaskForRegion(RegionCode aRegionCode);
```
Comment (bukepo) [inline src/common/region_code.hpp:48]: why not use the standard value(type char[2]) directly? https://en.wikipedia.org/wiki/ISO_3166-1.

File: src/common/region_code.hpp [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+enum RegionCode
+{
+    kRegionWW      = 0, // < World wide
+    kRegionCA      = 1, // < Canada
+    kRegionUS      = 2, // < United States
+    kRegionUnknown = 3, // < Unknown
+};
+
+RegionCode StringToRegionCode(const char *aRegionString);
```
Comment (bukepo) [inline src/common/region_code.hpp:44]: Missing doc string.

**Review by bukepo — APPROVED** [review]

File: src/common/region_code.cpp [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+ *    SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *    INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *    CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *    ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *    POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "region_code.hpp"
+
+#include <string.h>
```
Comment (Moandor-y) [inline src/common/region_code.cpp:31]: The `string.h` seems unused?

Comment (gjc13) [reply src/common/region_code.cpp:31]: Done

**Review by Moandor-y — APPROVED** [review]
LGTM

**Comment by jwhui** [comment]
@gjc13 , can you help resolve conflicts?

**Comment by gjc13** [comment]
> @gjc13 , can you help resolve conflicts?

Rebased

**Review by jwhui — APPROVED** [review]

**Pull request merged**